plugins/guests/tinycore/cap/change_host_name.rb

```diff
@@ -4,7 +4,6 @@ module Cap
       class ChangeHostName
         def self.change_host_name(machine, name)
           if !machine.communicate.test("hostname | grep '^#{name}$'")
-            machine.communicate.sudo("sh -c 'echo \"#{name}\" > /etc/hostname'")
```

Comment:
I don't think so...
if you start a box with the config.vm.hostname "mybox" in the Vagrant file
you will have a box name of mybox
Then vagrant halt --> pur the config.vm.hostname line in comment  --> vagrant up
The hostname will be boot2docker again...

Still why are you asking? Did we miss something?